### PR TITLE
Standalone decision evaluation MT E2E tests

### LINKED ISSUE #14563: Add E2E multi-tenancy tests for DMN decisions
Like any other feature, we must ensure that it behaves correctly.

We must add E2E tests to show that DMN decisions can be evaluated in a multi-tenant cluster, where the decision can be owned by specific tenants. Other tenants should not be allowed to access this data.

We must ensure the following features:
- evaluate decision through API
- evaluate decision through business rule task

### PR DESCRIPTION
**Pull request by remcowesterhoud**
## Description

This PR adds E2E tests for standalone decision evaluation.

The issue also describes an E2E test for business rule tasks. I have not implemented this. IMO it is not a sensible E2E test, as they are evaluated during process instance execution. There is no end to end, as the user doesn't have to send any commands to evaluate a business rule tasks, except for creating a process instance which is already covered.
I did add an extra unit test to verify that tenant A cannot reference the decision id of tenant B in a business rule task.

## Related issues

closes #14563 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by koevskinikola** [comment]
@remcowesterhoud did you mean to reference #14563 here? :smile:

**Comment by remcowesterhoud** [comment]
@koevskinikola What do you mean, this is linked 🤡 🙃 

I was wondering why it didn't link the PR to the issue this morning 😄  Thanks

File: engine/src/test/java/io/camunda/zeebe/engine/processing/incident/BusinessRuleTaskIncidentTest.java [hunk 1]
```diff
@@ -399,4 +400,48 @@ public void shouldCreateIncidentOnBusinessRuleTaskForCustomTenant() {
     // then
     assertIncidentCreated(processInstanceKey, taskActivating.getKey(), tenantId);
   }
+
+  @Test
+  public void shouldCreateIncidentOnBusinessRuleTaskForDifferentTenant() {
+    // given
+    final String tenantId = "acme";
+    final String otherTenantId = "emca";
```
Comment (koevskinikola) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/incident/BusinessRuleTaskIncidentTest.java:408]: :+1: Ooooh, a new one! :smile:

**Review by koevskinikola — APPROVED** [review]
LGTM @remcowesterhoud :100: 

Should we merge the PR with the `release-8.3.0` branch or merge it later with `main`?

**Comment by remcowesterhoud** [comment]
We can merge it to the release branch. There still needs to be another PR merged anyways.

bors merge

**Pull request merged**